src/main/services/patcher/patcher-middleware.ts

```diff
@@ -0,0 +1,38 @@
+import { Middleware } from 'redux';
+import { Patcher } from './patcher';
+
+
+export type PatcherMiddleware<T> = Middleware<{}, T>;
+
+export type PatcherMiddlewareOptions<T, U=T, A=any> = {
+  transform?: (state: U) => T;
+  select?: (action: A) => boolean;
+};
+
+const _DefaultOptions: PatcherMiddlewareOptions<any> = {
+  transform: (state) => state,
+  select: () => true,
+};
+
+
+export function createPatcherMiddleware<T, A=any, U=T>(
+  patcher: Patcher<T>,
+  options: PatcherMiddlewareOptions<T, U, A> = _DefaultOptions,
+): PatcherMiddleware<U>  {
+  const transform = options.transform || _DefaultOptions.transform!;
+  const select = options.select || _DefaultOptions.select!;
```

Comment:
I assume wrapping " _DefaultOptions.select" as Boolean( _DefaultOptions.select) and removing the force unwrapping ("!") should solve this one 🤔